src/box/memtx_hash.cc

```diff
@@ -621,6 +625,42 @@ hash_read_view_create_iterator(struct index_read_view *base,
 	return hash_read_view_iterator_start(it, type, key, part_count);
 }
 
+/** Implementation of dump_sort_data memtx_index_read_view callback. */
+static int
+hash_read_view_dump_primary_key(
+	struct memtx_index_read_view *base, ssize_t tuple_count,
+	struct memtx_sort_data *msd, bool *have_more)
+{
+	struct hash_read_view *rv = (struct hash_read_view *)base;
+
+	/* Collect the data to save. */
+	struct tuple **buffer =
+		(struct tuple **)xcalloc(tuple_count, sizeof(*buffer));
+	ssize_t dumped = 0;
+	while (dumped != tuple_count) {
+		struct tuple **ptuple = light_index_view_iterator_get_and_next(
+			&rv->view, &rv->dump_iterator);
+		if (ptuple == NULL) {
+			*have_more = false;
+			/* Restart the iterator for the next snapshot. */
+			light_index_view_iterator_begin(&rv->view,
```

Comment:
Not good, the iterator could have been broken before we got to the sort data file dump.

Reply:
It's actually OK for the current implementation (since the iterator state only contains a single number - `slotpos`). Updated the comment to the field.